src/nupic/algorithms/Connections.cpp

```diff
@@ -358,9 +365,9 @@ const SynapseData &Connections::dataForSynapse(Synapse synapse) const {
   return synapses_[synapse];
 }
 
-UInt32 Connections::segmentFlatListLength() const { return (UInt32)segments_.size(); }
+Segment Connections::segmentFlatListLength() const { return (Segment)segments_.size(); }
```

Comment:
If I remember right, `Segment` is both a typedef and a class elsewhere.  I know the namespaces keep the compilers from getting confused but but that does not help people trying to remember what is what.  Or helping sort out the results from text searches.  I am not recommending any changes, just pointing this out.

Reply:
yes, users should not be too confused either: 
- connections already operates (typedefs) a Segment, so it's that one (Segment here = UInt) 
- the other class Segment is only used in old TP,Cells4, and proposed for removal